MirrorProvider/MirrorProvider.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -134,29 +139,26 @@ private HResult EndDirectoryEnumeration(Guid enumerationId)
                     ProjectedFileInfo fileInfo = activeEnumeration.Current;
 
                     DateTime now = DateTime.UtcNow;
-                    result = results.Add(
+                    bool addResult = results.Add(
                         fileName: fileInfo.Name,
-                        fileSize: (ulong)(fileInfo.IsDirectory ? 0 : fileInfo.Size),
+                        fileSize: fileInfo.IsDirectory ? 0 : fileInfo.Size,
                         isDirectory: fileInfo.IsDirectory,
-                        fileAttributes: fileInfo.IsDirectory ? (uint)FileAttributes.Directory : (uint)FileAttributes.Archive,
+                        fileAttributes: fileInfo.IsDirectory ? FileAttributes.Directory : FileAttributes.Archive,
                         creationTime: now,
                         lastAccessTime: now,
                         lastWriteTime: now,
                         changeTime: now);
 
-                    if (result == HResult.Ok)
+                    if (addResult == true)
                     {
                         entryAdded = true;
                         activeEnumeration.MoveNext();
                     }
                     else
                     {
-                        if (result == HResult.InsufficientBuffer)
+                        if (entryAdded)

```

Comment:
@wilbaker we no longer have the option of checking for InsufficentBuffer.

Can you advise on this change?

Reply:
Yep there was a slight change in behavior here.  The code can be simplified to:

```
                if (results.Add(...))
                {
                    entryAdded = true;
                    enumeration.MoveNext();
                }
                else
                {
                    if (entryAdded)
                    {
                        hr = HResult.Ok;
                    }
                    else
                    {
                        hr = HResult.InsufficientBuffer;
                    }

                    break;
                }
```

Reply:
Small but i simplified the 'if' to 
result = entryAdded ? HResult.Ok : HResult.InsufficientBuffer;
                        break;